tsdb/store.go

```diff
@@ -613,6 +614,27 @@ func (s *Store) IteratorCreators() influxql.IteratorCreators {
 	return a
 }
 
+func (s *Store) IteratorCreator(shards []meta.ShardInfo) (influxql.IteratorCreator, error) {
+	// Generate iterators for each node.
+	ics := make([]influxql.IteratorCreator, 0)
+	if err := func() error {
+		for _, shard := range shards {
+			ic := s.ShardIteratorCreator(shard.ID)
+			if ic == nil {
+				continue
+			}
+			ics = append(ics, ic)
+		}
+
+		return nil
+	}(); err != nil {
+		influxql.IteratorCreators(ics).Close()
+		return nil, err
+	}
+
+	return influxql.IteratorCreators(ics), nil
+}
```

Comment:
A couple of reasons. The first is that it meant I could take the `iteratorCreator` method from the `QueryExecutor` and have it be different depending on the underlying store. The `tsdb.Store` is a local store and this will create local iterators from the list of shards. The clustering implementation will do something else in order to ensure we don't make a new connection for each shard.

Then I thought it made sense in this type anyway.